### PR TITLE
Better dbt support in jupyterhub

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

Resolves https://github.com/cal-itp/data-infra/issues/2438

Cleans up Jupyter and adds necessary support for initial dbt work using a poetry-managed virtual environment. Also updates documentation and changes the CI to edit existing docs preview comments instead of creating a new one each commit.

I'd like to investigate https://github.com/yuvipanda/jupyterhub-ssh as a follow-up.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation
- [ ] agencies.yml

## How has this been tested?
Been testing locally; the real test will be deploying to production and having someone go through the dbt setup from scratch.

### CODE REVIEW

File: docs/analytics_tools/github_setup.md [hunk 1]
```diff
@@ -3,46 +3,47 @@
...
+  exit status 3
+  Please try entering the URL in your browser manually
+✓ Authentication complete.
+- gh config set -h github.com git_protocol ssh
+✓ Configured git protocol
+✓ Uploaded the SSH key to your GitHub account: /home/jovyan/.ssh/id_ed25519.pub
+✓ Logged in as atvaccaro
+```
 
 After completing the steps above be sure to complete the section below to persist your SSH key between sessions and enable extensions.
```
Comment (lauriemerrell) [inline docs/analytics_tools/github_setup.md:48]: Is this still required? (persist the SSH key etc.)

**Review by lauriemerrell — COMMENTED** [review]
Just one question. I'd defer to @charlie-costanzo or @SorenSpicknall on the actual JupyterHub updates

File: docs/analytics_tools/github_setup.md (hunk 1 above)
Comment (atvaccaro) [reply docs/analytics_tools/github_setup.md:48]: Yes, `profile.sh` does configure the key with keychain but if a `.bashrc` file exists, it'll ignore `profile.sh` unless you specifically source it (as best I can tell)

**Comment by atvaccaro** [comment]
Ok docs are broken because of the pandas 2.0 release; [python-gbd](https://github.com/googleapis/python-bigquery-pandas/blob/main/setup.py#L28) does not specify a maximum version, so we'll need to do it ourselves. siuba is not compatible yet but it looks like michael is [working on it](https://github.com/machow/siuba/pull/472).

File: .github/workflows/publish-docs.yml [hunk 2]
```diff
@@ -51,28 +53,18 @@ jobs:
         NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
         PR_NUMBER: ${{ github.event.number }}
 
-    - name: Add netlify link PR comment
-      uses: actions/github-script@v3
+    - uses: peter-evans/find-comment@v2
```
Comment (SorenSpicknall) [inline .github/workflows/publish-docs.yml:56]: Not actionable on this PR, but I've mentioned something similar before - we should try to find a mechanism to identify if an individually authored action goes stale or falls out of maintenance, so we don't end up scrambling for a replacement in that case.

File: docs/analytics_examples/warehouse_tutorial.md [hunk 3]
```diff
@@ -33,7 +33,7 @@ The queries represented in the following tutorial are as follows:
 The tools that we can use to answer them are:
 * [**Metabase**](metabase) - our business insights and dashboarding tool
 * [**SQL**](querying-sql-jupyterhub) - using JupyterHub cloud notebooks
-* [**Python**](jupyterhub) - using JupyterHub cloud notebooks
+* [**Python**](jupyterhub-intro) - using JupyterHub cloud notebooks
```
Comment (SorenSpicknall) [inline docs/analytics_examples/warehouse_tutorial.md:36]: Do we have externally maintained links (in Google docs, slide decks, etc.) that are going to need updating from this?

**Review by SorenSpicknall — APPROVED** [review]

File: docs/analytics_examples/warehouse_tutorial.md (hunk 3 above)
Comment (atvaccaro) [reply docs/analytics_examples/warehouse_tutorial.md:36]: Hard to know for sure but unlikely, we tend to just link to the top-level page

**Comment by atvaccaro** [comment]
I just added back the portfolio dependency group; I'd removed when trying to install dbt in the image directly, but we can keep it if we're using `poetry` for warehouse work.

**Pull request merged**